test/unit/integrated_function_test.C

```diff
@@ -0,0 +1,274 @@
+        {
+          libMesh::Real L = 10.0/std::cos(theta[t]);
+          libMesh::Real sintheta = std::sin(theta[t]);
+          libMesh::Real costheta = std::cos(theta[t]);
+
+          std::vector<std::string> functions;
+          std::vector<libMesh::Real> calc_answers;
+
+          // constant
+          functions.push_back("1");
+          calc_answers.push_back(L);
+
+          // linear
+          functions.push_back("x");
+          calc_answers.push_back(0.5/costheta*10.0*10.0);
+          functions.push_back("y");
+          calc_answers.push_back( sintheta/2.0*L*L + 2.6*L );
+
+          // polynomial
+          functions.push_back("x*(y^2)");
+          calc_answers.push_back(0.25*std::pow(L,4)*std::pow(sintheta,2)*costheta + (2.0/3.0)*std::pow(L,3)*costheta*sintheta*2.6 + 0.5*L*L*costheta*2.6*2.6);
+          functions.push_back("(4/3)*(x^3)+10");
+          calc_answers.push_back( (4.0/12.0)*pow(costheta,3)*std::pow(L,4)+10*L );
+
+          GRINS::MultiphysicsSystem* system = sim.get_multiphysics_system();
+
+          // build rayfire
+          GRINS::SharedPtr<GRINS::RayfireMesh> rayfire = new GRINS::RayfireMesh(origin,theta[t]);
+
+          CPPUNIT_ASSERT_EQUAL(functions.size(), calc_answers.size());
+
+          // and now the CompositeQoI to do the evalutaion
+          GRINS::CompositeQoI comp_qoi;
+
+          for (unsigned int i=0; i<functions.size(); i++)
+            {
+              GRINS::IntegratedFunction<libMesh::FunctionBase<libMesh::Real> > integ_func((unsigned int)2,new libMesh::ParsedFunction<libMesh::Real>(functions[i]),rayfire,"integrated_function");
+              comp_qoi.add_qoi(integ_func);
+            }
+
+          comp_qoi.init(input,*system);
+          system->attach_qoi(&comp_qoi);
+          system->assemble_qoi();
+
+          for (unsigned int i=0; i<calc_answers.size(); i++)
+              CPPUNIT_ASSERT_DOUBLES_EQUAL( calc_answers[i], sim.get_qoi_value(i),libMesh::TOLERANCE  );
+
+        } // for t
+
+        // Clear out the VariableWarehouse so it doesn't interfere with other tests.
+        GRINS::GRINSPrivate::VariableWarehouse::clear();
+    }
+
+    //! These functions cannot be integrated exactly wih Gauss Quadrature,
+    //! and so we look for linear convergence to the analytical solution
+    void test_convergence()
+    {
+      const std::string filename = std::string(GRINS_TEST_UNIT_INPUT_SRCDIR)+"/integrated_function_quad9.in";
+      GetPot input( filename );
+      const char* const argv = "unit_driver";
+      GetPot empty_command_line( (const int)1,&argv );
+
+      libMesh::Point origin(0.0,2.6);
+
+      // angles for rayfire
+      std::vector<libMesh::Real> theta;
+      theta.push_back(0.0);
+      theta.push_back(0.25);
+
+      GRINS::SimulationBuilder sim_builder;
+
+      GRINS::Simulation sim(input,
+                            empty_command_line,
+                            sim_builder,
+                            *TestCommWorld );
+
```

Comment:
Actually, you can probably just override `setUp()` (I think that's the name) and that will be called before each test. Similarly for `tearDown()` (where you could put the `VariableWarehouse` clearing.